packages/core/src/components/graphs/bar.ts

```diff
@@ -1,16 +1,21 @@
 // Internal Imports
 import { Component } from "../component";
+import { DOMUtils } from "../../services";
 
 export class Bar extends Component {
 	// Gets the correct width for bars based on options & configurations
 	protected getBarWidth(allDataLabels?: string[]) {
 		const options = this.model.getOptions();
+		const numberOfDatapoints = this.model.getDisplayData().length;
 		const mainXScale = this.services.cartesianScales.getMainXScale();
+		const chartWidth = DOMUtils.getSVGElementSize(this.parent, {
+			useAttrs: true
+		}).width;
 
 		if (!mainXScale.step) {
 			return Math.min(
 				options.bars.maxWidth,
-				(5 / mainXScale.ticks().length) * options.bars.maxWidth
+				chartWidth * 0.25 / numberOfDatapoints
```

Comment:
how did you get to this calculation? could you explain?

Reply:
Yes. The maximum width for the bars in the chart can be `width of the chart / numberOfDatapoints` so that they don't overlap with each other. But we need to leave some spaces between the bars and the beginning and end of the chart, which is why I multiply by 0.25 here. How I determined this number was sort of trial and error with various different data. With dense data of minutes of difference (just like my demo), even seconds, any number greater than this would result in overlapping. This seams to work for all the data for me. The original logic in determining was wrong and could result in bar width greater than the chart width.